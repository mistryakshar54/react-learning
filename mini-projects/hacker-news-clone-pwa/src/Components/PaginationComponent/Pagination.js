import React from 'react';
import Paper from "@material-ui/core/Paper";
import IconButton from "@material-ui/core/IconButton";
import KeyboardArrowLeft from "@material-ui/icons/KeyboardArrowLeft";
import KeyboardArrowRight from "@material-ui/icons/KeyboardArrowRight";

const PaginationComponent = ( props ) => {

    const { currentPage ,totalPages , nextPage } = props;

    return (
      <Paper style={{ marginTop: "5%" }} elevation={3}>
        <div>
          <IconButton
            onClick={() => {
              nextPage("previous");
            }}
            disabled={currentPage < 1}
          >
            {" "}
            <KeyboardArrowLeft />{" "}
          </IconButton>
          {currentPage} / {totalPages}
          <IconButton
            onClick={() => {
              nextPage("next");
            }}
            disabled={currentPage >= totalPages}
          >
            <KeyboardArrowRight />
          </IconButton>
        </div>
      </Paper>
    );
}

export default PaginationComponent;