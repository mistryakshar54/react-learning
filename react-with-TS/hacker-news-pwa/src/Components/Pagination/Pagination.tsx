import React from 'react';
import Paper from "@material-ui/core/Paper";
import IconButton from "@material-ui/core/IconButton";
import KeyboardArrowLeft from "@material-ui/icons/KeyboardArrowLeft";
import KeyboardArrowRight from "@material-ui/icons/KeyboardArrowRight";

type PaginationProps = {
    currentPage : number,
    nextPage : (direction : string) =>void
}
const Pagination = ({ currentPage, nextPage }: PaginationProps) => {
    const totalPages = 17;
    return (
        <Paper style={{ marginTop: "5%" }} elevation={3}>
            <div>
                <IconButton
                    onClick={() => {
                        nextPage("previous");
                    }}
                    disabled={currentPage < 1}
                >
                    {" "}
                    <KeyboardArrowLeft />{" "}
                </IconButton>
                {currentPage} / {totalPages}
                <IconButton
                    onClick={() => {
                        nextPage("next");
                    }}
                    disabled={currentPage >= totalPages}
                >
                    <KeyboardArrowRight />
                </IconButton>
            </div>
        </Paper>
    );
}

export default Pagination;