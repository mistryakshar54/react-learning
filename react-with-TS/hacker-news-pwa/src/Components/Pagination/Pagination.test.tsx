import * as React from "react";
import Enzyme,{mount} from 'enzyme';
import Pagination from "./Pagination";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";

Enzyme.configure({ adapter: new Adapter() });

it("renders snapshot", () => {
    const component = renderer.create(
        <Pagination currentPage={1} nextPage={() => {}} />
    );
    expect(component).toMatchSnapshot();
});