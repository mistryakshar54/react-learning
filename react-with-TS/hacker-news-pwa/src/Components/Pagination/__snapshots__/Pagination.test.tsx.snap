// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation3 MuiPaper-rounded"
  style={
    Object {
      "marginTop": "5%",
    }
  }
>
  <div>
    <button
      className="MuiButtonBase-root MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
         
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
          />
        </svg>
         
      </span>
    </button>
    1
     / 
    17
    <button
      className="MuiButtonBase-root MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;
